### PR TITLE
Add Boyer-Moore string search algorithm

### PR DESCRIPTION
**Pull request by ayDavidGitHere**
Add New Algorithm.

### CODE REVIEW

**Comment by raklaptudirm** [comment]
Fix the raised alerts.

File: String/BoyerMoore.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+let boyerMoore = (str, pattern)=> {
+  var badMatchTable = buildBadMatchTable(pattern),
+  offset = 0,
+  patternLastIndex = pattern.length - 1,
+  scanIndex = patternLastIndex,
+  maxOffset = str.length - pattern.length;
+  // if the offset is bigger than maxOffset, cannot be found
+  while (offset <= maxOffset) {
```
Comment (appgurueu) [inline String/BoyerMoore.js:33]: Please fix your indentation.

Comment (ayDavidGitHere) [reply String/BoyerMoore.js:31]: I just completed the fixes.

Comment (appgurueu) [reply String/BoyerMoore.js:31]: You still didn't fix the indentation.

Comment (ayDavidGitHere) [reply String/BoyerMoore.js:31]: The last commit, this 40bd41c3be55e44b8cd1d704aa56913bce24c254 did.
It passed the code style check & the lgtm check

File: String/BoyerMoore.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    while (pattern[scanIndex] === str[scanIndex + offset]) {
+      if (scanIndex === patternLastIndex) {
+        // found at this index
+        return offset
+      }
+      scanIndex++
+    }
+    const badMatchString = str[offset + patternLastIndex]
+    if (badMatchTable[badMatchString]) {
+    // increase the offset if it exists
```
Comment (appgurueu) [inline String/BoyerMoore.js:42]: Please indent this comment

File: String/BoyerMoore.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      offset += badMatchTable[badMatchString]
+    } else {
+      offset += 1
```
Comment (appgurueu) [inline String/BoyerMoore.js:45]: `offset++`

**Review by appgurueu — APPROVED** [review]
Otherwise LGTM

**Comment by raklaptudirm** [comment]
@ayDavidGitHere Waiting for @appgurueu 's comments to be resolved.

**Review by appgurueu — APPROVED** [review]

**Review by raklaptudirm — CHANGES_REQUESTED** [review]
Format your code with `standard.js`.

**Review by raklaptudirm — APPROVED** [review]

**Review by appgurueu — APPROVED** [review]

**Pull request merged**